### ISSUE TITLE
Lavalink fails to download when /tmp and /data are on different drives

### ISSUE THREAD

**Issue by dafzor** [Cog-Creators/Red-DiscordBot#2714]
# Other bugs

#### What were you trying to do?

Running red-discordbot on a docker container with data mounted on a docker volume. Using this template https://github.com/PhasecoreX/docker-red-discordbot

#### What were you expecting to happen?

lavalink to download normally and be booted by the Audio cog.

#### What actually happened?

download fails with the following error:

```
> [p]load audio
[2019-05-20 19:20:30] [INFO] red.audio.manager: Downloading Lavalink.jar...
> [p]llsetup external
> [p]llsetup external
[2019-05-20 19:23:56] [INFO] red.audio.manager: Downloading Lavalink.jar...
[2019-05-20 19:24:24] [ERROR] red.audio: Connecting to Lavalink server timed out, retrying...
[2019-05-20 19:24:25] [INFO] red.audio.manager: Downloading Lavalink.jar...
Task exception was never retrieved
future: <Task finished coro=<Audio.attempt_connect() done, defined at /data/venv/lib/python3.7/site-packages/redbot/cogs/audio/audio.py:123> exception=OSError(18, 'Invalid cross-device link')>
Traceback (most recent call last):
  File "/data/venv/lib/python3.7/site-packages/redbot/cogs/audio/audio.py", line 136, in attempt_connect
    await self._manager.start()
  File "/data/venv/lib/python3.7/site-packages/redbot/cogs/audio/manager.py", line 62, in start
    await self.maybe_download_jar()
  File "/data/venv/lib/python3.7/site-packages/redbot/cogs/audio/manager.py", line 243, in maybe_download_jar
    await cls._download_jar()
  File "/data/venv/lib/python3.7/site-packages/redbot/cogs/audio/manager.py", line 216, in _download_jar
    pathlib.Path(path).replace(LAVALINK_JAR_FILE)
  File "/usr/local/lib/python3.7/pathlib.py", line 1321, in replace
    self._accessor.replace(self, target)
OSError: [Errno 18] Invalid cross-device link: '/tmp/tmpgkgk1im6' -> '/data/cogs/Audio/Lavalink.jar'
```

#### How can we reproduce this issue?

Using ubuntu 18.04 LTS with docker install the above container.

additional information: https://github.com/PhasecoreX/docker-red-discordbot/issues/5

**Comment by PhasecoreX** [comment]
From what little research I've done on this so far, doing file moves using shutil.move would support cross-device paths. Also the /tmp directory is usually on its own file system (tmpfs) compared to the rest of the system, so I'm curious as to how this was working originally not in a docker container.

But again, haven't looked into any of this yet.

**Comment by mikeshardmind** [comment]
Hey, thanks for the report. We should probably be swapping usage of `os.rename` for `shutil.move` except in explicit cases where we don't want it to auto-determine issues.

Until then, you should be able to set an environment variable in the docker image for the path 
`TMPDIR=/some/path/on/same/device`

**Comment by Tobotimus** [comment]
This was fixed in #2765 :smiley:
